physics/Interstitials/UFS_SCM_NEPTUNE/GFS_photochemistry.F90

```diff
@@ -0,0 +1,105 @@
+module GFS_photochemistry
+  use machine,        only: kind_phys
+  use module_ozphys,  only: ty_ozphys
+  use module_h2ophys, only: ty_h2ophys
+  implicit none
+contains
+
+! #########################################################################################
+!> \section arg_table_GFS_photochemistry_init Argument Table
+!! \htmlinclude GFS_photochemistry_init.html
+!!
+! #########################################################################################
+  subroutine GFS_photochemistry_init(oz_phys_2006, oz_phys_2015, h2o_phys, errmsg, errflg)
+    logical, intent(in) :: &
+         oz_phys_2015, & ! Do ozone photochemistry? (2015)
+         oz_phys_2006, & ! Do ozone photochemistry? (2006)
+         h2o_phys        ! Do stratospheric h2o photochemistry?
+    character(len=*), intent(out) :: &
+         errmsg          ! CCPP Error message.
+    integer,  intent(out) :: &
+         errflg          ! CCPP Error flag.
+
+    ! Initialize CCPP error handling variables
+    errmsg = ''
+    errflg = 0
+
+    ! If no photchemical scheme is on, but SDF has this module, report an error?
+    ! DJS2024 Asks: If everything in _run() is controlled by these logicals, we probably don't
+    !         need to kill it here, since the run phase will be harmless?
+    if ((.not. oz_phys_2006) .and. (.not. oz_phys_2015) .and. (.not. h2o_phys)) then
+       write (errmsg,'(*(a))') 'Logic error: One of [oz_phys_2006, oz_phys_2015, or h2o_phys] must == .true. '
+       errflg = 1
+       return
+    endif
+
+  end subroutine GFS_photochemistry_init
+
+! #########################################################################################
+!> \section arg_table_GFS_photochemistry_run Argument Table
+!! \htmlinclude GFS_photochemistry_run.html
+!!
+! #########################################################################################
+  subroutine GFS_photochemistry_run (dtp, ozphys, oz_phys_2015, oz_phys_2006, con_1ovg,   &
+       prsl, dp, ozpl, h2o_phys, h2ophys, h2opl, h2o0, oz0, gt0, do3_dt_prd, do3_dt_ozmx, &
+       do3_dt_temp, do3_dt_ohoz, errmsg, errflg)
+    
+    ! Inputs
+    real(kind=kind_phys), intent(in) :: &
+         dtp,          & ! Model timestep
+         con_1ovg        ! Physical constant (1./gravity)
+    real(kind=kind_phys), intent(in), dimension(:,:) :: &
+         prsl,         & ! Air pressure (Pa)
+         dp,           & ! Pressure thickness (Pa)
+         gt0             ! Air temperature (K)
+    real(kind=kind_phys), intent(in), dimension(:,:,:) :: &
+         ozpl,         & ! Ozone data for current model timestep.
+         h2opl           ! h2o data for curent model timestep.
+    logical,              intent(in) :: &
+         oz_phys_2015, & ! Do ozone photochemistry? (2015)
+         oz_phys_2006, & ! Do ozone photochemistry? (2006)
+         h2o_phys        ! Do stratospheric h2o photochemistry?
+    type(ty_ozphys), intent(in) :: &
```

Comment:
Are ozphys and/or h2ophys conditionally allocated, i.e. do they need to be optional here?

Reply:
They are required by the scheme and are not optional from the physics POV.